File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int maximum = 0, ans = 1, count = 0;
        
        // Find the maximum element
        for(int num : nums) {
            maximum = max(maximum, num);
        }
        
        // Count the longest subarray of maximum element
        for(int num : nums) {
            if(num == maximum) {
                ans = max(ans, ++count);
            } else {
                count = 0;
            }
        }
        
        return ans;
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maximum = max(nums)  
        ans = 1
        count = 0
        
        for num in nums:
            if num == maximum:
                count += 1
                ans = max(ans, count)
            else:
                count = 0
                
        return ans
